pkg/util/openstack/loadbalancer.go

```diff
@@ -668,13 +668,17 @@ func CreateL7Rule(client *gophercloud.ServiceClient, policyID string, opts l7pol
 }
 
 // UpdateHealthMonitor updates a health monitor.
-func UpdateHealthMonitor(client *gophercloud.ServiceClient, monitorID string, opts monitors.UpdateOpts) error {
+func UpdateHealthMonitor(client *gophercloud.ServiceClient, monitorID string, opts monitors.UpdateOpts, lbID string) error {
 	mc := metrics.NewMetricContext("loadbalancer_healthmonitor", "update")
 	_, err := monitors.Update(client, monitorID, opts).Extract()
 	if mc.ObserveRequest(err) != nil {
 		return fmt.Errorf("failed to update healthmonitor: %v", err)
 	}
 
+	if _, err := WaitActiveAndGetLoadBalancer(client, lbID); err != nil {
+		return fmt.Errorf("failed to wait for load balancer %s ACTIVE after updating healthmonitor: %v", lbID, err)
+	}
```

Comment:
Hmm, this is a bad controller pattern for at least 2 reasons:
* It ties up a controller thread waiting for a single LB to become active, which artificially limits concurrency.
* It takes no account of of a controller restart

Ideally, if an operation requires the LB to be active, we should check it immediately before whatever requires it be active and return from Reconcile() if it's not ready yet.

However, as this pattern seems to be idiomatic in this code base it's probably best to fix that all at once as a separate PR.

Reply:
I agree with you on this one, a while ago I was thinking we should change how we handle LB statuses, especially as with Amphora it can be 10s of seconds before LB is `ACTIVE` again. This patch only fixes the current state of things in which we could try listener creation on an LB that's still being `PENDING_UPDATE` because of the health monitor update. Such call would inevitably fail.